trin-history/src/utils.rs

```diff
@@ -0,0 +1,41 @@
+use trin_core::portalnet::types::{HistoryContentKey, HeaderKey, BodyKey, ReceiptsKey};
+use trin_core::portalnet::U256;
+use sha3::{Digest, Sha3_256};
+use std::convert::TryInto;
+
+// content-id derivation function for history network
+pub fn history_network_derive_content_id(key: &String) -> U256 {
+    // decode key
+    let decoded_key = match HistoryContentKey::from_bytes(&key.as_bytes()) {
+        Ok(k) => k,
+        Err(_) => panic!("Could not decode key"),
+    };
+
+    // calculate content-id
+    match decoded_key {
+        HistoryContentKey::HeaderKey(HeaderKey { chain_id, content_type, block_hash }) => {
+            let input = U256::from(chain_id) + U256::from(content_type) + block_hash;
```

Comment:
It's not immediately obvious to me that using `+` on a U256 will concatenate the bytes instead of numerically add the results. In fact, I would have guessed the opposite, but I guess that's just more of a complaint with the U256 lib. If there's another way to be more explicit that we are concatenating bytes, just for readability in trin, I think that would be preferable.

Reply:
I was under the impression that the `+` was adding the U256s, which is what I thought we wanted? I'm not sure since the spec is not clear on this point. I also add two U256s in calculating the ContractStorageTrieProof key, specified by the spec [here](https://github.com/ethereum/portal-network-specs/blob/master/state-network.md#contract-storage-trie-proof). The only thing I have to support that `+` is actually addition and not concatenation is that in the state-network spec `|` is used for concatenation, but `+` symbol is used, so I assumed its addition. History spec is unclear.

Reply:
Oh, interesting! I just hadn't noticed that it was actually addition in some places. Will U256 automatically roll-over as the spec defines? More things to test. I'm also not clear why we are sometimes using addition and sometimes concatenation. There are fewer question marks with concat (like not having to worry about rollover), so I would recommend pushing back on the spec a little, if there's no obviously good reason to use addition.